[CI] SnapshotBasedIndexRecoveryIT testRecoveryIsCancelledAfterDeletingTheIndex failing

This is a weird timeout issue. From what I can tell there are no related issues. Seems suspect that its on arm.

**Build scan:**
https://gradle-enterprise.elastic.co/s/exqmn7xel4bcg/tests/:server:internalClusterTest/org.elasticsearch.indices.recovery.SnapshotBasedIndexRecoveryIT/testRecoveryIsCancelledAfterDeletingTheIndex

**Reproduction line:**
`./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.indices.recovery.SnapshotBasedIndexRecoveryIT.testRecoveryIsCancelledAfterDeletingTheIndex" -Dtests.seed=E36E1308685B0AF6 -Dtests.locale=th-TH-u-nu-thai-x-lvariant-TH -Dtests.timezone=Indian/Reunion -Druntime.java=16`

**Applicable branches:**
7.x

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.indices.recovery.SnapshotBasedIndexRecoveryIT&tests.test=testRecoveryIsCancelledAfterDeletingTheIndex

**Failure excerpt:**
```
java.lang.Exception: Test abandoned because suite timeout was reached.

  at __randomizedtesting.SeedInfo.seed([E36E1308685B0AF6]:0)

```

So, it may not be due to arm. 

Had another failure: https://gradle-enterprise.elastic.co/s/y3rabg37c5mxm

Still linux, and still a timeout waiting for cluster green state for `java.lang.AssertionError: timed out waiting for green state for index [snapshot_based_recovery] cluster state [{`

--- a/server/src/internalClusterTest/java/org/elasticsearch/indices/recovery/SnapshotBasedIndexRecoveryIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/indices/recovery/SnapshotBasedIndexRecoveryIT.java
@@ -475,12 +475,6 @@ public class SnapshotBasedIndexRecoveryIT extends AbstractSnapshotIntegTestCase 
             createSnapshot(repoName, "snap", Collections.singletonList(indexName));
 
             String targetNode = internalCluster().startDataOnlyNode();
-            assertAcked(
-                client().admin().indices().prepareUpdateSettings(indexName)
-                    .setSettings(Settings.builder()
-                        .put("index.routing.allocation.require._name", targetNode)).get()
-            );
-
             MockTransportService targetMockTransportService =
                 (MockTransportService) internalCluster().getInstance(TransportService.class, targetNode);
 
@@ -494,6 +488,12 @@ public class SnapshotBasedIndexRecoveryIT extends AbstractSnapshotIntegTestCase 
                     respondToRecoverSnapshotFile.await();
                     handler.messageReceived(request, channel, task);
                 }
+            );
+
+            assertAcked(
+                client().admin().indices().prepareUpdateSettings(indexName)
+                    .setSettings(Settings.builder()
+                        .put("index.routing.allocation.require._name", targetNode)).get()
             );
 
             recoverSnapshotFileRequestReceived.await();

